azext_edge/tests/helpers.py

```diff
@@ -7,13 +7,115 @@
 import json
 import os
 import shutil
+from fnmatch import fnmatch
 from knack.log import get_logger
-from typing import Dict
+from typing import Any, Dict, List, Optional, Union
 from azure.cli.core.azclierror import CLIInternalError
+import pytest
+
+from azext_edge.edge.providers.edge_api.base import EdgeResourceApi
 
 logger = get_logger(__name__)
 
 
+def filter_resources(
+    kubectl_items: List[Dict[str, Any]],
+    prefixes: Optional[Union[str, List[str]]] = None,
+    resource_match: Optional[str] = None,
+) -> Dict[str, Any]:
+    def pass_conditions(item: Dict[str, Any]) -> bool:
+        name = item["metadata"]["name"]
+        if resource_match and not fnmatch(name, resource_match):
```

Comment:
can change into a single return

`return any([
   not resource_match and not prefixes,
   resource_match and fnmatch(name, resource_match),
   prefixes and any(name.startswith(prefix) for prefix in prefixes)
])`